### ISSUE TITLE
XSite fails with 3 or more sites on the same cluster

### ISSUE THREAD

**Issue by ryanemerson** [infinispan/infinispan-operator#1834]
```yaml
apiVersion: infinispan.org/v1
kind: Infinispan
metadata:
  name: keycloak-infinispan
spec:
  replicas: 3
  version: 14.0.9
  service:
    type: DataGrid
    sites:
      local:
        name: EUROPE
        expose:
          type: ClusterIP
        maxRelayNodes: 100
      locations:
        - name: ASIA
          url: infinispan+xsite://infinispan-cs.keycloak-asia.svc.cluster.local:7900
        - name: US
          url: infinispan+xsite://infinispan-cs.keycloak-us.svc.cluster.local:7900
  security:
    authorization:
      enabled: true
    endpointSecretName: connect-secret
    endpointEncryption:
      type: None
  logging:
    categories:
      org.infinispan: trace
      org.jgroups.protocols.TCP: error
      org.jgroups.protocols.relay.RELAY2: error
  container:
    cpu: 2000m:100m
    extraJvmOpts: -XX:NativeMemoryTracking=summary
    memory: 2Gi:256Mi
```

The user was able to workaround the issue by downgrading to the 2.2.x Operator,. Could this related to the reconciliation of the GossipRouter?

[operator.tar.gz](https://github.com/infinispan/infinispan-operator/files/12111431/operator.tar.gz)
[infinispan.tar.gz](https://github.com/infinispan/infinispan-operator/files/12111432/infinispan.tar.gz)

**Comment by dmmishyn** [comment]
This issue is still exists with operator version 2.4.2

**Comment by ryanemerson** [comment]
@dmmishyn The issue should now be resolved from 2.4.4
